pkg/controller/cassandradatacenter/reconciler.go

```diff
@@ -57,20 +70,38 @@ func (reconciler *CassandraDataCenterReconciler) Reconcile(request reconcile.Req
 
 	nodesService, err := createOrUpdateNodesService(rctx)
 	if err != nil {
+		log.Error(err, "Error building nodes service")
 		return reconcile.Result{}, err
 	}
 
 	seedNodesService, err := createOrUpdateSeedNodesService(rctx)
+	if err != nil {
+		log.Error(err, "Error building seeds service")
+		return reconcile.Result{}, err
+	}
+
+	// check if we need to do anything at all, that is the number of nodes matches running ones
+	allPods, err := AllPodsInCDC(rctx.client, cdc)
+	if int32(len(allPods)) == cdc.Spec.Nodes {
+		log.Info("All pods are running, nothing to reconcile")
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		log.Error(err, "can't find all pods, skipping reconcile cycle")
+		return reconcile.Result{}, err
+	}
+
+	// fine, so we have a mismatch. Will seek to reconcile.
+	rackSpec, err := getRackSpec(rctx, request)
```

Comment:
Yes, we now only reconcile one rack per reconcile. You are correct that at the moment it doesn't support config changes/upgrades, but scaling up/down will work because k8 will reconcile until the total number of pods matches the spec.

Reply:
Thanks @zegelin.

1. I will look into dropping the rack/dc data into the pod/C* container and configuring it either via script or snitch. This will eliminate the need for multiple config maps and will get us a bit closer to doing it all with 1 StatefulSet (not yet possible really).
2. Yes, `Rack` is better, I only recently introduced it as a convenience and haven't gotten to fixing all the places yet. Will do so.
3. We don't need to loop over racks to reconcile because we have it for free from k8 which will call the Reconciler until all the racks (StatefulSets) are reconciled. So we only care for finding any 1 rack that needs reconciling and do that.

Also, at the moment all this wouldn't support upgrades or conf changes (such as different image url/version or hardware changes or whatever), as if there is no change in replica numbers, nothing will be done. I will remove that check and will see how much work needed for taking care of that (considering that for some of those changes we would need rolling restarts of the C* containers).  Should we support it for the "first cut"?